### DISCUSSION TITLE
Yield Today for Battery powered inverters

### DISCUSSION THREAD

**Discussion by Duz91** [tbnobody/OpenDTU#1748] [Ideas]
Hallo liebe OpenDTU Community, ich liebe euer tool und wollte mich für die professionelle Bereitstellung und Pflege bedanken. Ich hätte eine kleine kosmetische Idee. Die Wechselrichter sind eigentlich ausgelegt für Solarpanele. Dementsprechend ist es leicht eine "Nacht" zu erkennen und den "Yiel Today Counter" zurückzusetzen. Für Batteriebetriebene Wechselrichter würde ich mir eine Softwarelösung wünschen. Ich weiss nicht wo der Yield Today Counter läuft ... wenn er auf dem Wechselrichter läuft der sich nur durch Spannungslosigkeit zurücksetzen lässt könnte man ggf einen zweiten Energiezähler einrichten, der nur lokal auf dem OpenDTU aufintegriert wird. Natürlich könnte ich auch in Homeassistant einen neustart des Wechselrichters antriggern aber ich weiss nicht, ob das meine Hardware irgendwann schädigt.

Gruß
David

**Comment by MetaChuh** [comment]
hallo @Duz91 

> ... neustart des Wechselrichters antriggern aber ich weiss nicht, ob das meine Hardware irgendwann schädigt.

das schädigt den inverter nicht.
es ist der gleiche vorgang, inklusive relais trennung und aktivierung, wie wenn ein solar inverter sich bei sonnenuntergang ausschaltet und bei sonnenaufgang wieder einschaltet (nur mit weniger pause dazwischen)

den batterie inverter via api um 23:59h oder 00:00h zu restarten, ist die beste und empfohlene vorgangsweise.

> Ich weiss nicht wo der Yield Today Counter läuft.

der wert `YieldDay` kommt direkt aus dem inverter.

> ... ggf einen zweiten Energiezähler einrichten, der nur lokal auf dem OpenDTU aufintegriert wird.

dieser ansatz wurde schon oft überlegt, jedoch erwies sich ein offset subtrahieren/addieren beim schwester dtu projekt, wo dies getestet wird, bis heute als massiv problematisch.

wenn zb. die dtu rebootet wird, und dadurch der offset nicht mehr vorhanden ist, springt der `YieldDay` wert sofort auf den komplettwert seit letzter dc power trennung, was jegliche aufzeichnungen und auswertungen ab dem zeitpunkt extrem und dauerhaft verfälscht.
(zb. plötzlich einen tagesertrag von 60000wh um 10:00h früh, der sich dann in deiner statistik einbrennt)

zb. dtu updates dürftest du dann nur genau um mitternacht machen, und nur nachdem du den inverter ebenfalls rebootet hast.
sonst beginnt die dtu mit 0 offset, jedoch der inverter liefert alle kwh seit letzem reboot.

hoffe die info ist hilfreich

greetings
metachuh

**Reply by Duz91** [reply]
Vielen Dank für die schnelle und freundliche Antwort! Habe mir gedacht, dass das alles seine Gründe hat. 
Wäre ein sauberer Workaround ein geplanter Neustart aus OpenDTU heraus? Das müsste doch ziemlich leicht umzusetzen sein ggf in Verbindung mit dem Herabsetzen der Leistung kurz vor dem Neustart (hier wieder der Gedanke HW schonen). 

Aber dann schaue ich primär, wie ich eine Home Assistant Automatisation bzgl Neustart machen kann.

**Reply by MetaChuh** [reply]
> wie ich eine Home Assistant Automatisation bzgl Neustart machen kann.

gibt viele möglichkeiten.
zb. mit einem home assistant `shell_command` um mitternacht

der curl befehl dazu:
(funktioniert auch über jedes linux, mac, windows terminal, oder cron, task planner)

```bash
curl --user "admin:your_password" --request POST --data "data={'serial':'your_battery_inverter_serial','restart':true}" http://10.10.10.77/api/power/config
```

manche senden den reboot, oder andere befehle, auch bare metal von ihren shelly 3em oder plugs weg.
(via zeitschaltuhr, oder action script)

> Wäre ein sauberer Workaround ein geplanter Neustart aus OpenDTU heraus?

ja, `restart inverter at midnight` wäre ein nettes feature.
ich müsste mal beim fork OpenDTU-OnBattery nachschauen, ich bilde mir ein, dort gibts das schon.

**Comment by spcqike** [comment]
Hallo,

alternativ kannst du auch den Fork "openDTU-onBattery" ausprobieren. dort lässt sich ein geplanter Neustart des Wechselrichters hinterlegen.

https://github.com/helgeerbe/OpenDTU-OnBattery

**Reply by MetaChuh** [reply]
steht schon oben, aber doppelt hält besser 😉
thx

**Reply by spcqike** [reply]
Das hab ich irgendwie überlesen :o

**Reply by MetaChuh** [reply]
no prob 😄
und danke dir für die bestätigung,
ich wusste nicht mehr auswendig, ob inverter reboot bei OpenDTU-OnBattery schon eingebaut war 🙏
